### PR TITLE
Add `uniqueValues` operator

### PR DESCRIPTION
**Pull request by mdiep**
Supersedes #2715 based on my feedback.

### CODE REVIEW

File: ReactiveCocoaTests/Swift/SignalSpec.swift [hunk 1]
```diff
@@ -660,7 +660,37 @@ class SignalSpec: QuickSpec {
 				expect(disposedItems) == [ true, false, false, true ]
 			}
 		}
+		
+		describe("uniqueValues", {
```
Comment (ikesyo) [inline ReactiveCocoaTests/Swift/SignalSpec.swift]: A trailing closure should be used here.

File: ReactiveCocoa/Swift/Signal.swift [hunk 2]
```diff
@@ -1216,6 +1216,57 @@ extension SignalType {
...
 
+extension SignalType {
+	/// Forwards only those values from `self` that have unique identities across the set of
+	/// all values that have been seen.
+	///
+	/// Note: This causes the identities to be retained to check for uniqueness.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func uniqueValues<Identity: Hashable>(transform: Value -> Identity) -> Signal<Value, Error> {
+		return Signal { observer in
+			let seenValues: Atomic<Set<Identity>> = Atomic([])
```
Comment (ikesyo) [inline ReactiveCocoa/Swift/Signal.swift]: Is the `Atomic` use really needed? It looks like the set is not accessed or modified concurrently in the implementation.

File: ReactiveCocoaTests/Swift/SignalSpec.swift [hunk 3]
```diff
@@ -660,7 +660,37 @@ class SignalSpec: QuickSpec {
...
+			it("should skip values that have been already seen"){
```
Comment (ikesyo) [inline ReactiveCocoaTests/Swift/SignalSpec.swift]: Could you add a space before `{`?

File: ReactiveCocoa/Swift/Signal.swift [hunk 4]
```diff
@@ -1216,6 +1216,57 @@ extension SignalType {
...
+								var mutableSet = set
+								mutableSet.insert(identity)
+								return mutableSet
+							}
+						}
+						if isUnique {
+							fallthrough
+						}
+						
+					default:
```
Comment (ikesyo) [inline ReactiveCocoa/Swift/Signal.swift]: `case .Failed, .Completed, .Interrupted:` would be preferable as with `skipWhile`.

**Comment by mdiep** [comment]
Great feedback, @ikesyo! I've addressed it. ✨

File: ReactiveCocoa/Swift/Signal.swift [hunk 5]
```diff
@@ -1216,6 +1216,47 @@ extension SignalType {
...
+	}
+}
+
+extension SignalType where Value: Hashable {
+	/// Forwards only those values from `self` that are unique across the set of
+	/// all values that have been seen.
+	///
+	/// Note: This causes the values to be retained to check for uniqueness. Providing
+	/// a function that returns a unique value for each sent value can help you reduce
+	/// the memory footprint.
```
Comment (ikesyo) [inline ReactiveCocoa/Swift/Signal.swift]: 👍 for the documentation.

**Comment by ikesyo** [comment]
Looks great, thank you @nixterrimus and @mdiep!

**Pull request merged**